### PR TITLE
Add notes with regards to enabling service catalog

### PR DESCRIPTION
**Pull request by cdrage**
Adds some notes with regards to actually enabling service catalog for
development and getting started.

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -90,6 +90,17 @@ Using project "myproject".
 
 Now we can move on to creating our application using `odo`.
 
+
+#### Notes with regards to Service Catalog / `odo service` functionality:
+
+In order to make **full use** of all Odo functionality, it's recommended to enable [service catalog](https://docs.openshift.com/container-platform/3.6/architecture/service_catalog/index.html) with OpenShift.
```
Comment (kadel) [inline docs/getting-started.md]: 3.6 is really really old version. It would be better to refer to latest stable - 3.11

File: docs/development.md [hunk 2]
```diff
@@ -186,7 +186,12 @@ All tests can be found in the `tests/e2e` directory and can be called using func
 
 Requirements:
 
- - A `minishift` or OpenShift environment with Service Catalog enabled
+ - A `minishift` or OpenShift environment with Service Catalog enabled$a
```
Comment (kadel) [inline docs/development.md]: what is `$a`?

File: docs/getting-started.md (hunk 1 above)
Comment (kadel) [reply docs/getting-started.md]: it looks like you can use https://docs.openshift.com/container-platform/latest/architecture/service_catalog/index.html that will redirect to the latest version automaticaly

File: docs/development.md (hunk 2 above)
Comment (geoand) [reply docs/development.md]: Typo I guess?

File: docs/getting-started.md (hunk 1 above)
Comment (surajnarwade) [reply docs/getting-started.md]: :+1:

**Comment by cdrage** [comment]
@kadel @geoand Yeah, that was a `vim` typo. PR updated! (removed typo + updated to 3.11 version)

**Comment by kadel** [comment]
ignoring circleCi check, this is just a docs change

**Pull request merged**